lib/document_viewer_web/ensure_document_viewer_group.ex

```diff
@@ -0,0 +1,28 @@
+defmodule DocumentViewerWeb.EnsureDocumentViewerGroup do
+  @moduledoc """
+  Verify that the user is in our Cognito group.
+  """
+
+  import Plug.Conn
+
+  def init(options), do: options
+
+  def call(conn, opts) do
+    with %{"groups" => groups} <- Guardian.Plug.current_claims(conn),
+         true <- is_list(groups),
+         document_viewer_group <-
+           Keyword.get(
+             opts,
+             :document_viewer_group,
+             Application.get_env(:document_viewer, :cognito_group)
+           ),
```

Comment:
I was thinking more like:
```elixir
def init(opts), do:
  Keyword.put_new(
    options, 
    :document_viewer_group, 
    Application.get_env(:document_viewer, :cognito_group))
end

def call(conn, opts):
  ...
  document_viewer_group = Keyword.fetch!(opts, :document_viewer_group),
  ...
end
```

Reply:
Oh sure, thanks for the clarification. That makes a lot of sense - I fixed it up.